src/nvim/ops.c

```diff
@@ -2524,6 +2528,61 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, long y_idx)
   *pnew = NUL;
 }
 
+static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+{
+  char buf[NUMBUFLEN+2];
+
+  if (!has_event(EVENT_TEXTYANKPOST)) {
+    // No autocommand was defined
+    return;
+  }
+
+  // set v:event to a dictionary with information about the yank
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
```

Comment:
> because somebody set e.g. v:event.regcontents

The dictionary values should be read-only, no user can set them. And a test for this would be good.